src/SignalR/common/Http.Connections/src/ConnectionOptionsSetup.cs

```diff
@@ -16,16 +16,23 @@ public class ConnectionOptionsSetup : IConfigureOptions<ConnectionOptions>
         /// </summary>
         public static TimeSpan DefaultDisconectTimeout = TimeSpan.FromSeconds(15);
 
+        internal static readonly TimeSpan DefaultShutdownDelay = TimeSpan.FromSeconds(1);
```

Comment:
What should the default be?

Reply:
I wonder if we can get this from the host configuration? If not, we can assume it's 5 seconds.

Reply:
Issue with it being 5 seconds is that the host shutdown timeout will be hit and race with Kestrel closing connections and SignalR closing connections.